File: 938_Range_Sum_of_BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        self.ans=0
        def helper(node):
            if node:
                if(node.val>=low and node.val<=high):
                    self.ans+=node.val
                if(low<node.val):
                    helper(node.left)
                if(node.val<high):
                    helper(node.right)
                    
        helper(root)
        return (self.ans)
        
        

File: 938_Range_Sum_of_BST.cpp
class Solution {
public:
    int rangeSumBST(TreeNode* root, int low, int high) {
        int sum = 0;
        if(root == NULL) return 0;
        if(root->val >= low && root->val <= high)
        {
            sum+= root->val;
        }
        if(root->val > high)
        {
            sum += rangeSumBST(root->left,low,high);
        }
        else if(root->val < low)
        {
            sum+= rangeSumBST(root->right,low,high);
        }
        else{
            sum+= rangeSumBST(root->left,low,high) + rangeSumBST(root->right,low,high);
        }
        
        
        return sum;
        
    }
};